pkg/backend/cancellation_scope.go

```diff
@@ -49,16 +50,19 @@ func (cancellationScopeSource) NewScope(events chan<- engine.Event, isPreview bo
 
 	c := &cancellationScope{
 		context: cancelContext,
-		sigint:  make(chan os.Signal),
+		sigint:  make(chan os.Signal, 1),
 		done:    make(chan bool),
 	}
 
 	go func() {
-		for range c.sigint {
-			// If we haven't yet received a SIGINT, call the cancellation func. Otherwise call the termination
-			// func.
+		for sig := range c.sigint {
+			// If we haven't yet received a SIGINT or SIGTERM, call the cancellation func. Otherwise call the
+			// termination func.
 			if cancelContext.CancelErr() == nil {
 				message := "^C received; cancelling. If you would like to terminate immediately, press ^C again.\n"
+				if sig == syscall.SIGTERM {
+					message = "SIGTERM received; cancelling. If you would like to terminate immediately, send SIGTERM again.\n"
```

Comment:
I updated the message here, but I remember I looked into system messages (`engine.NewEvent(engine.StdoutEventPayload{...})`) at some time before, and I never managed to get any to display. I am not sure they ever show up I the current UI.